parser/src/main.rs

```diff
@@ -112,7 +112,9 @@ where
         match self {
             Ok(t) => t,
             Err(e) => {
-                tracing::debug!("Error details: {:#?}", e);
+                // QUESTION: Since we're expecting to parse this output as logs,
```

Comment:
Yeah, I think this seems like a good idea. At a minimum, it would help to standardize as much as possible on field names for things like filename and line number in the error representations.

Reply:
I've moved the error formatting to tracing's field syntax, rather than directly interpolating them into the message. 

<details>
  <summary>Examples of tracing output:</summary>

```jsonl
{
  "timestamp": "Oct 01 11:22:56.814",
  "level": "WARN",
  "fields": {
    "message": "failed to parse row",
    "error": "Parse(InvalidContent(Error(UnequalLengths { pos: Some(Position { byte: 664, line: 62, record: 61 }), expected_len: 2, len: 3 })))"
  },
  "target": "parser::format::csv",
  "span": {
    "parse_args": "ParseArgs { config_file: Some(\"/tmp/jsonl-parser-test.SaI2L21nhmbp/config.json\"), file: \"-\"}",
    "name": "do_parse"
  },
  "spans": [
    {
      "parse_args": "ParseArgs { config_file: Some(\"/tmp/jsonl-parser-test.SaI2L21nhmbp/config.json\"), file: \"-\" }",
      "name": "do_parse"
    }
  ]
}
{
  "timestamp": "Oct 01 11:22:56.815",
  "level": "DEBUG",
  "fields": {
    "error_details": "ErrorLimitExceeded(ErrorThreshold { max_percent: 25 })",
    "message": "parsing failed"
  },
  "target": "parser",
  "span": {
    "parse_args": "ParseArgs { config_file: Some(\"/tmp/jsonl-parser-test.SaI2L21nhmbp/config.json\"), file: \"-\" }",
    "name": "do_parse"
  },
  "spans": [
    {
      "parse_args": "ParseArgs { config_file: Some(\"/tmp/jsonl-parser-test.SaI2L21nhmbp/config.json\"), file: \"-\" }",
      "name": "do_parse"
    }
  ]
}
{
  "timestamp": "Oct 01 11:22:56.815",
  "level": "ERROR",
  "fields": {
    "error": "error limit exceeded",
    "message": "parsing failed"
  },
  "target": "parser",
  "span": {
    "parse_args": "ParseArgs { config_file: Some(\"/tmp/jsonl-parser-test.SaI2L21nhmbp/config.json\"), file: \"-\" }",
    "name": "do_parse"
  },
  "spans": [
    {
      "parse_args": "ParseArgs { config_file: Some(\"/tmp/jsonl-parser-test.SaI2L21nhmbp/config.json\"), file: \"-\" }",
      "name": "do_parse"
    }
  ]
}
```
</details>

This is the minimal amount of json formatting. The errors themselves are still the default `Debug` format for Rust structs/enums. If we want to make these errors even easier to parse, I think we'll want to go back and implement `Serialize` on our error types.